src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2367,6 +2380,16 @@ private void doSignalWriteIdleTimeoutIfNecessary()
             }
         }
 
+        private void doSignalCloseTimeout()
+        {
+            closeTimeoutAt = System.currentTimeMillis() + CLOSE_TIMEOUT;
+
+            if (closeTimeoutId == NO_CANCEL_ID)
+            {
+                closeTimeoutId = signaler.signalAt(closeTimeoutAt, routeId, replyId, CLOSE_SIGNAL_ID);
```

Comment:
We should cancel this `closeTimeoutId` if not  equal to `NO_CANCEL_ID` when `CLOSE` frame is decoded, agree?

Reply:
yes